javaLoginCore/src/main/java/com/hamdan/forzenbook/java/login/core/data/repository/LoginRepository.java

```diff
@@ -0,0 +1,9 @@
+package com.hamdan.forzenbook.java.login.core.data.repository;
+
+import androidx.annotation.Nullable;
+
+public interface LoginRepository {
+    public void requestValidation(String email);
+
+    public User getToken(@Nullable String email, @Nullable String code) throws NullTokenException, FailTokenRetrievalException;
```

Comment:
Why are these nullable?

Reply:
because theres two possible outcomes of this call, in Kotlin it was originaly two similar functions with different parameters. So we merged it into one. Would it be better for this to be two different functions in Java?